pkg/externalplugin/types.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package externalplugin
+
+// PluginRequest contains all information kubebuilder received from the CLI
+// and plugins executed before it.
+type PluginRequest struct {
+	// Command contains the command to be executed by the plugin such as init, create api, etc.
+	Command string `json:"command"`
+
+	// APIVersion defines the versioned schema of PluginRequest that is being sent from Kubebuilder.
+	// Initially, this will be marked as alpha (v1alpha1).
+	APIVersion string `json:"apiVersion"`
+
+	// Args holds the plugin specific arguments that are received from the CLI
+	// which are to be passed down to the external plugin.
+	Args []string `json:"args"`
+
+	// Universe represents the modified file contents that gets updated over a series of plugin runs
+	// across the plugin chain. Initially, it starts out as empty.
+	Universe map[string]string `json:"universe"`
```

Comment:
Does this preclude binary files?

Reply:
do we support binary files -->

Not that I know of (atleast for the first pass).  In case we want to support binary files in the future (for example JPEG image), we can switch this to a map of string and byte slice. 
```suggestion
	Universe map[string][]byte `json:"universe"`
```

I do know that we want to support post-plugin hooks that are executables as a part of `PluginResponse` struct, as follow-up implementation after the initial implementation is done.

Reply:
binary files are not always executables. There are many binary files. A gzipped xml file could be a binary file. A jar file would be a binary file. What if my plugin wants to create a jar file and output that to the universe? How would I write the contents of this binary file to the JSON to return to kubebuilder?

Reply:
Okay, looks like JSON doesn't support "binary" natively. So it would have to be base64 encoded which is strings. https://www.json.org/json-en.html So I think `map[string]string` would work.